tooling/metamodel-generator/src/main/java/org/hibernate/processor/validation/ProcessorSessionFactory.java

```diff
@@ -578,7 +578,8 @@ private static boolean isMatchingEntity(Element symbol, String entityName) {
 		if (symbol.getKind() == ElementKind.CLASS) {
 			final TypeElement type = (TypeElement) symbol;
 			return isEntity(type)
-				&& getJpaEntityName(type).equals(entityName);
+				   && (getJpaEntityName( type ).equals( entityName )
+					   || type.getQualifiedName().toString().equals( entityName ));
```

Comment:
```suggestion
				&& ( getJpaEntityName(type).equals(entityName)
					|| type.getQualifiedName().contentEquals(entityName) );
```

Reply:
It seems that this is OK, but I will check before pushing change.